src/Uno.UI.Composition/Composition/ContainerVisual.skia.cs

```diff
@@ -25,6 +26,13 @@ partial void InitializePartial()
 		{
 			IsChildrenRenderOrderDirty = true;
 
+			var parent = this;
+			while (parent is not null && parent._subtreeVisualCount is not null)
+			{
+				parent._subtreeVisualCount = null;
```

Comment:
Wouldn't it be more performant if you adjust this based on the given NotifyCollectionChangedAction?

- Remove: subtract one
- Add: add one
- Replace/Move: do nothing
- Reset: set back to null to re-calculate

Reply:
That's fair but the average number of children of a visual is likely around 2. Looping through 3 children and reading a cached value is hardly more expensive and it leaves the code a lot cleaner and easier to debug when we make invalidation bugs. We can revisit this if it shows up while profiling.